src/basic/fileio.c

```diff
@@ -780,6 +781,7 @@ static int merge_env_file_push(
 
         if (!env_name_is_valid(key)) {
                 log_error("%s:%u: invalid variable name \"%s\", ignoring.", strna(filename), line, key);
+                free(value);
```

Comment:
Actually, I didn't read https://github.com/systemd/systemd/pull/5131 So, I don't fully understand what does this code do :-)